fuzz/fuzz-util.sh

```diff
@@ -12,7 +12,8 @@ targetFileToName() {
   echo "$1" \
     | sed 's/^fuzz_targets\///' \
     | sed 's/\.rs$//' \
-    | sed 's/\//_/g'
+    | sed 's/\//_/g' \
+    | sed 's/^_//g'
```

Comment:
Had to add this because the script was generating target file names that started with `_` and was failing the `diff` in `verify-execution` step of the fuzz CI.

Reply:
I am having a hard time running `honggfuzz` in my Mac Apple Silicon:

```shell
Caused by:
  process didn't exit successfully: `/Users/einherjar/git/btc/rust-bitcoin/hfuzz_target/release/build/honggfuzz-a49925690b981dda/build-script-build` (exit status: 101)
  --- stderr
  Makefile:103: *** Unsupported MAC OS X version.  Stop.
  thread 'main' panicked at /Users/einherjar/.cargo/registry/src/index.crates.io-6f17d22bba15001f/honggfuzz-0.5.55/build.rs:41:5:
  assertion failed: status.success()
```

Can you post the diff you are getting by running it locally?

Reply:
```diff
diff --git a/.github/workflows/fuzz.yml b/.github/workflows/fuzz.yml
index ba802c787..06e02b5cd 100644
--- a/.github/workflows/fuzz.yml
+++ b/.github/workflows/fuzz.yml
@@ -1,61 +1,61 @@
 # Automatically generated by fuzz/generate-files.sh
 name: Fuzz
 
 on:
   push:
     branches:
       - master
       - 'test-ci/**'
   pull_request:
 
 jobs:
   fuzz:
     if: ${{ !github.event.act }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         fuzz_target: [
-          hashes_sha1,
-          hashes_sha256,
-          hashes_sha512,
-          hashes_json,
-          hashes_cbor,
           hashes_ripemd160,
+          hashes_cbor,
           hashes_sha512_256,
-          bitcoin_deserialize_prefilled_transaction,
+          hashes_json,
+          hashes_sha1,
+          hashes_sha512,
+          hashes_sha256,
+          bitcoin_deserialize_psbt,
           bitcoin_deserialize_amount,
+          bitcoin_deserialize_script,
+          bitcoin_deserialize_transaction,
+          bitcoin_script_bytes_to_asm_fmt,
+          bitcoin_deserialize_prefilled_transaction,
           bitcoin_deser_net_msg,
+          bitcoin_deserialize_block,
+          bitcoin_outpoint_string,
           bitcoin_deserialize_address,
-          bitcoin_script_bytes_to_asm_fmt,
           bitcoin_deserialize_witness,
-          bitcoin_outpoint_string,
-          bitcoin_deserialize_transaction,
-          bitcoin_deserialize_psbt,
-          bitcoin_deserialize_script,
-          bitcoin_deserialize_block,
         ]
     steps:
       - name: Install test dependencies
         run: sudo apt-get update -y && sudo apt-get install -y binutils-dev libunwind8-dev libcurl4-openssl-dev libelf-dev libdw-dev cmake gcc libiberty-dev
       - uses: actions/checkout@v4
       - uses: actions/cache@v3
         id: cache-fuzz
         with:
           path: |
             ~/.cargo/bin
             fuzz/target
             target
           key: cache-${{ matrix.target }}-${{ hashFiles('**/Cargo.toml','**/Cargo.lock') }}
       - uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: '1.65.0'
       - name: fuzz
         run: |
           if [[ "${{ matrix.fuzz_target }}" =~ ^bitcoin ]]; then
               export RUSTFLAGS='--cfg=hashes_fuzz --cfg=secp256k1_fuzz'
           fi
           echo "Using RUSTFLAGS $RUSTFLAGS"
           cd fuzz && ./fuzz.sh "${{ matrix.fuzz_target }}"
       - run: echo "${{ matrix.fuzz_target }}" >executed_${{ matrix.fuzz_target }}
       - uses: actions/upload-artifact@v3
diff --git a/fuzz/Cargo.toml b/fuzz/Cargo.toml
index 3a251b43c..dafa0f349 100644
--- a/fuzz/Cargo.toml
+++ b/fuzz/Cargo.toml
@@ -1,89 +1,89 @@
 [package]
 name = "bitcoin-fuzz"
 edition = "2018"
 version = "0.0.1"
 authors = ["Generated by fuzz/generate-files.sh"]
 publish = false
 
 [package.metadata]
 cargo-fuzz = true
 
 [dependencies]
 honggfuzz = { version = "0.5.55", default-features = false }
 bitcoin = { version = "0.30.0", features = [ "serde" ] }
 
 serde = { version = "1.0.103", features = [ "derive" ] }
 serde_json = "1.0"
 serde_cbor = "0.9"
 
 [[bin]]
-name = "hashes_sha1"
-path = "fuzz_targets//hashes/sha1.rs"
+name = "hashes_ripemd160"
+path = "fuzz_targets/hashes/ripemd160.rs"
 
 [[bin]]
-name = "hashes_sha256"
-path = "fuzz_targets//hashes/sha256.rs"
+name = "hashes_cbor"
+path = "fuzz_targets/hashes/cbor.rs"
 
 [[bin]]
-name = "hashes_sha512"
-path = "fuzz_targets//hashes/sha512.rs"
+name = "hashes_sha512_256"
+path = "fuzz_targets/hashes/sha512_256.rs"
 
 [[bin]]
 name = "hashes_json"
-path = "fuzz_targets//hashes/json.rs"
+path = "fuzz_targets/hashes/json.rs"
 
 [[bin]]
-name = "hashes_cbor"
-path = "fuzz_targets//hashes/cbor.rs"
+name = "hashes_sha1"
+path = "fuzz_targets/hashes/sha1.rs"
 
 [[bin]]
-name = "hashes_ripemd160"
-path = "fuzz_targets//hashes/ripemd160.rs"
+name = "hashes_sha512"
+path = "fuzz_targets/hashes/sha512.rs"
 
 [[bin]]
-name = "hashes_sha512_256"
-path = "fuzz_targets//hashes/sha512_256.rs"
+name = "hashes_sha256"
+path = "fuzz_targets/hashes/sha256.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_prefilled_transaction"
-path = "fuzz_targets//bitcoin/deserialize_prefilled_transaction.rs"
+name = "bitcoin_deserialize_psbt"
+path = "fuzz_targets/bitcoin/deserialize_psbt.rs"
 
 [[bin]]
 name = "bitcoin_deserialize_amount"
-path = "fuzz_targets//bitcoin/deserialize_amount.rs"
+path = "fuzz_targets/bitcoin/deserialize_amount.rs"
 
 [[bin]]
-name = "bitcoin_deser_net_msg"
-path = "fuzz_targets//bitcoin/deser_net_msg.rs"
+name = "bitcoin_deserialize_script"
+path = "fuzz_targets/bitcoin/deserialize_script.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_address"
-path = "fuzz_targets//bitcoin/deserialize_address.rs"
+name = "bitcoin_deserialize_transaction"
+path = "fuzz_targets/bitcoin/deserialize_transaction.rs"
 
 [[bin]]
 name = "bitcoin_script_bytes_to_asm_fmt"
-path = "fuzz_targets//bitcoin/script_bytes_to_asm_fmt.rs"
+path = "fuzz_targets/bitcoin/script_bytes_to_asm_fmt.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_witness"
-path = "fuzz_targets//bitcoin/deserialize_witness.rs"
+name = "bitcoin_deserialize_prefilled_transaction"
+path = "fuzz_targets/bitcoin/deserialize_prefilled_transaction.rs"
 
 [[bin]]
-name = "bitcoin_outpoint_string"
-path = "fuzz_targets//bitcoin/outpoint_string.rs"
+name = "bitcoin_deser_net_msg"
+path = "fuzz_targets/bitcoin/deser_net_msg.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_transaction"
-path = "fuzz_targets//bitcoin/deserialize_transaction.rs"
+name = "bitcoin_deserialize_block"
+path = "fuzz_targets/bitcoin/deserialize_block.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_psbt"
-path = "fuzz_targets//bitcoin/deserialize_psbt.rs"
+name = "bitcoin_outpoint_string"
+path = "fuzz_targets/bitcoin/outpoint_string.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_script"
-path = "fuzz_targets//bitcoin/deserialize_script.rs"
+name = "bitcoin_deserialize_address"
+path = "fuzz_targets/bitcoin/deserialize_address.rs"
 
 [[bin]]
-name = "bitcoin_deserialize_block"
-path = "fuzz_targets//bitcoin/deserialize_block.rs"
+name = "bitcoin_deserialize_witness"
+path = "fuzz_targets/bitcoin/deserialize_witness.rs"
```

Reply:
Sorry, I am not following.
Working on @vincenzopalazzo suggestions:

> To really fix the script you should generate the ci from the script and not change the CI and then update the script, otherwise, there is not reason to have the script.

`.github/workflows/fuzz.yml` and `fuzz/Cargo.toml` was generated automatically from the `fuzz/generate-files.sh`.

Reply:
When I run `fuzz/generate-files.sh` I get different output than what's committed.

Reply:
That explains the

> Now when I run the script locally I get extra `/`s and things get reordered.

I think that `fuzz/generate-files.sh` had not been run in a while in the repo?

Reply:
We can do something like

```diff
diff --git a/fuzz/fuzz-util.sh b/fuzz/fuzz-util.sh
index 7761c38d..759d41c4 100755
--- a/fuzz/fuzz-util.sh
+++ b/fuzz/fuzz-util.sh
@@ -4,7 +4,7 @@ REPO_DIR=$(git rev-parse --show-toplevel)
 
 listTargetFiles() {
   pushd "$REPO_DIR/fuzz" > /dev/null || exit 1
-  find fuzz_targets/ -type f -name "*.rs"
+  find fuzz_targets/ -type f -name "*.rs" | sort
   popd > /dev/null || exit 1
 }
``` 

That will make the list of generated files stable.

Reply:
Lol, I had thought we'd already done this.

@realeinherjar definitely ACK sticking `| sort` onto anything that lists files.

Reply:
solved in 01909529ee5eb35b78a6f826b406e963607b29f9